### PR TITLE
🌱 (ci): Improve GitHub Action to linter samples under testdata

### PR DESCRIPTION
**Pull request by camilamacedo86**
This commit change the action in order to use a matrix to validate all samples under testdata

### CODE REVIEW

**Comment by camilamacedo86** [comment]
c/c @mateusoliveira43 
Related to the changes proposed at: https://github.com/kubernetes-sigs/kubebuilder/pull/4425

File: .github/workflows/lint-sample.yml [hunk 1]
```diff
@@ -11,6 +11,13 @@ on:
 jobs:
   lint-samples:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        folder: [
```
Comment (mateusoliveira43) [inline .github/workflows/lint-sample.yml:16]: there are other test projects in docs folder, right?

Should they be added here as well?

**Comment by camilamacedo86** [comment]
@mateusoliveira43

Could you LGTM this one?
Then, rebase your PR https://github.com/kubernetes-sigs/kubebuilder/pull/4425 with this one?

In the PR https://github.com/kubernetes-sigs/kubebuilder/pull/4425 we need add the config + ensure that we call it for Kubebuilder CLI + samples. WDYT?

**Review by mateusoliveira43 — APPROVED** [review]
I can rebase my PR afterwards :)

File: .github/workflows/lint-sample.yml (hunk 1 above)
Comment (camilamacedo86) [reply .github/workflows/lint-sample.yml:16]: We are not linting the docs samples for the tutorial
We probably should to do that as well
We need an issue for it. See: https://github.com/kubernetes-sigs/kubebuilder/issues/4445

**Pull request merged**